src/plugins/mvcc_delete_plugin.cpp

```diff
@@ -0,0 +1,177 @@
+#include "mvcc_delete_plugin.hpp"
+
+#include "operators/get_table.hpp"
+#include "operators/table_wrapper.hpp"
+#include "operators/update.hpp"
+#include "operators/validate.hpp"
+#include "storage/pos_list.hpp"
+#include "storage/reference_segment.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+MvccDeletePlugin::MvccDeletePlugin()
+    : _sm(StorageManager::get()),
+      _delete_threshold_rate_invalidated_rows(0.8),
+      _delete_threshold_commit_diff_factor(1.5),
+      _idle_delay_logical_delete(std::chrono::milliseconds(1000)),
+      _idle_delay_physical_delete(std::chrono::milliseconds(1000)) {}
+
+const std::string MvccDeletePlugin::description() const { return "Physical MVCC delete plugin"; }
+
+void MvccDeletePlugin::start() {
+  _loop_thread_logical_delete = std::make_unique<PausableLoopThread>(_idle_delay_logical_delete,
+                                                                     [this](size_t) { this->_logical_delete_loop(); });
+
+  _loop_thread_physical_delete = std::make_unique<PausableLoopThread>(
+      _idle_delay_physical_delete, [this](size_t) { this->_physical_delete_loop(); });
+}
+
+void MvccDeletePlugin::stop() {
+  // Call destructor of PausableLoopThread to terminate its thread
+  _loop_thread_logical_delete.reset();
+  _loop_thread_physical_delete.reset();
+}
+
+/**
+ * This function analyzes each chunk of every table and triggers a chunk-cleanup-procedure if a certain threshold of invalidated rows is exceeded.
```

Comment:
That line looks a bit long. Have you used the formater?

Reply:
Formatter didn't change anything, but you're right, it looks too long